plugin/metadata/metadata_test.go

```diff
@@ -47,15 +47,45 @@ func TestMetadataServeDNS(t *testing.T) {
 	}
 
 	ctx := context.TODO()
+	if IsMetadataSet(ctx) {
+		t.Errorf("Context provide Metadata information whereas metadata is not yet activated")
+	}
 	m.ServeDNS(ctx, &test.ResponseWriter{}, new(dns.Msg))
 	nctx := next.ctx
 
+	if !IsMetadataSet(nctx) {
+		t.Errorf("Context does not provide Metadata information")
+	}
+
 	for _, expected := range expectedMetadata {
 		for label, expVal := range expected {
+			if !IsLabel(label) {
+				t.Errorf("Expected label %s is not considered a valid label", label)
+			}
 			val := ValueFunc(nctx, label)
 			if val() != expVal() {
 				t.Errorf("Expected value %s for %s, but got %s", expVal(), label, val())
 			}
 		}
 	}
 }
+
+func TestLabelFormat(t *testing.T) {
+	labels := []struct {
+		label   string
+		isValid bool
+	}{
+		{"plugin/LABEL", true},
```

Comment:
`/` would default to true as well and is invalid. The IsLabel function can fairly easy be extended to count the character to the index of / and check if we see another slash in the next part of the string.